proposals/function-pointers.md

```diff
@@ -274,39 +289,105 @@ We use `System.Runtime.InteropServices.OutAttribute`, applied as a `modreq` to t
-`System.Runtime.CompilerServices` namespace in the core library are considered to be part of the calling convention of the
-function pointer.
+C# recognizes 4 special identifiers that map to specific existing unmanaged `CallKind`s from ECMA 335. In order for this
+mapping to occur, these identifiers must be specified on their own, with no other identifiers, and this requirement is
+encoded into the spec for `unmanaged_calling_convention`s. These identifiers are `Cdecl`, `Thiscall`, `Stdcall`, and
+`Fastcall`, which correspond to `unmanaged cdecl`, `unmanaged thiscall`, `unmanaged stdcall`, and `unmanaged fastcall`,
+respectively. If more than one `identifer` is specified, or the single `identifier` is not of the specially recognized
+identifiers, we perform special name lookup on the identifier with the following rules:
 
-### UnmanagedCallersOnlyAttribute
+* We prepend the `identifier` with the string `CallConv`
+* We look only at types defined in the `System.Runtime.CompilerServices` namespace.
+* We look only at types defined in the core library of the application, which is the library that defines `System.Object`
+and has no dependencies.
 
-`UnmanagedCallersOnlyAttribute` is an attribute used by the CLR to indicate that a method should be called with a specific
-calling convention. Because of this, we introduce the following support for working with the attribute:
+If lookup succeeds on all of the `identifier`s specified in an `unmanaged_calling_convention`, we encode the `CallKind` as
+`unmanaged ext`, and encode each of the resolved types in the set of `modopt`s at the beginning of the function pointer
+signature. As a note, these rules mean that users cannot prefix these `identifier`s with `CallConv`, as that will result
+in looking up `CallConvCallConvVectorCall`.
+
+When interpreting metadata, the we first look at the `CallKind`. If it is anything other than `unmanaged ext`, we ignore all
+`modopt`s on the return type for the purposes of determining the calling convention, and use only the `CallKind`. If the
+`CallKind` is `unmanaged ext`, we look at the modopts at the start of the function pointer type, taking the union of all
+types that meet the following requirements:
+
+* The is defined in the core library, which is the library that references no other libraries and defines `System.Object`.
+* The type is defined in the `System.Runtime.CompilerServices` namespace.
+* The type starts with the prefix `CallConv`.
+
+ These represent the types that must be found when performing lookup on the `identifier`s in an
+`unmanaged_calling_convention` when defining a function pointer type in source.
+
+It is an error to attempt to use a function pointer with a `CallKind` of `unmanaged ext` if the target runtime does not
+support the feature. This will be determined by looking for the presence of the
+`System.Runtime.CompilerServices.RuntimeFeature.UnmanagedCallKind` constant. If this constant is present, the runtime is
```

Comment:
I thought the presence of the `UnmanagedCallersOnlyAttribute` was going to be enough here?

Reply:
I see the note below.

Reply:
I don't particularly want to condition the support based on that attribute, as I think they're orthogonal and we shouldn't tie them together, but we can discuss that Thursday in design review :).